src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java

```diff
@@ -0,0 +1,278 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2015 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.indentation;
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * This Check controls the indentation inside comments.
+ * There're should be whitespace between comment declaration (//, /*)
+ * and it's content. Single-line block comments (/* ... *&#47;)
+ * have to have a whitespace before the {@link TokenTypes#BLOCK_COMMENT_END block comment end}
+ * (*&#47;)
+ * <p>
+ * Examples:
+ * </p>
+ * <p>
+ * Check's config:
+ * </p>
+ * <p>
+ * <code>
+ * &lt;module name=&quot;CommentsIndentation&quot;/&gt;
+ * </code>
+ * </p>
```

Comment:
Would be good to have some idents for readablity
```xml
<p>
  Examples:
</p>
<p>
  Check's config:
</p>
<p>
  <code>
    &lt;module name=&quot;CommentsIndentation&quot;/&gt;
  </code>
</p>
```
but it would cause this check to fail on itself. Just a thing to consider.